src/containers/Application/ApplicationSubmissionNEW.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react'
+import { Button, Header, Icon, Label, List, Message, Segment } from 'semantic-ui-react'
+import evaluate from '@openmsupply/expression-evaluator'
+import Markdown from '../../utils/helpers/semanticReactMarkdown'
+import { ApplicationProps, EvaluatorParameters } from '../../utils/types'
+import { useUserState } from '../../contexts/UserState'
+import { useRouter } from '../../utils/hooks/useRouter'
+import strings from '../../utils/constants'
+import { Link } from 'react-router-dom'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+
+const ApplicationSubmission: React.FC<ApplicationProps> = ({ structure }) => {
+  const [submissionMessageEvaluated, setSubmissionMessageEvaluated] = useState('')
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const {
+    query: { serialNumber },
+    push,
+  } = useRouter()
+  const {
+    info: { current, submissionMessage, type },
+    stages,
+  } = structure
+
+  // Check if application not submitted and redirect to the summary page
+  // Note: The summary page has its own redirection logic to a specific page (with invalid items).
+  if (
+    current?.status === ApplicationStatus.Draft ||
+    current?.status === ApplicationStatus.ChangesRequired
+  )
+    push(`/applicationNEW/${serialNumber}/summary`)
+
+  const evaluatorParams: EvaluatorParameters = {
+    objects: { currentUser },
+    APIfetch: fetch,
+  }
+  evaluate(submissionMessage || '', evaluatorParams).then((result: any) =>
+    setSubmissionMessageEvaluated(result)
+  )
```

Comment:
Do you think we should evaluate this in `useLoadApplication`, to keep more logic out of this display component?

Reply:
I'll have a thought tomorrow, I think it's a good idea, but wonder if should do the same for `startMessage` in `useLoadTemplate` => In PR #404 - XD

Reply:
Yes, I actually just made the same comment on that one too -- haven't submitted review yet though.